packages/cli/src/lib/history/history-command.unit.test.ts

```diff
@@ -0,0 +1,119 @@
+import { describe, expect, vi } from 'vitest';
+import { type HistoryOptions, history, readRcByPath } from '@code-pushup/core';
+import { safeCheckout } from '@code-pushup/utils';
+import { DEFAULT_CLI_CONFIGURATION } from '../../../mocks/constants';
+import { yargsCli } from '../yargs-cli';
+import { yargsHistoryCommandObject } from './history-command';
+
+vi.mock('@code-pushup/core', async () => {
+  const {
+    MINIMAL_HISTORY_CONFIG_MOCK,
+  }: typeof import('@code-pushup/test-utils') = await vi.importActual(
+    '@code-pushup/test-utils',
+  );
+  const core: object = await vi.importActual('@code-pushup/core');
+  return {
+    ...core,
+    history: vi
+      .fn()
+      .mockImplementation((options: HistoryOptions, commits: string[]) =>
+        commits.map(commit => `${commit}-report.json`),
+      ),
+    readRcByPath: vi.fn().mockResolvedValue(MINIMAL_HISTORY_CONFIG_MOCK),
+  };
+});
+
+vi.mock('@code-pushup/utils', async () => {
+  const utils: object = await vi.importActual('@code-pushup/utils');
+  return {
+    ...utils,
+    safeCheckout: vi.fn(),
+    getCurrentBranchOrTag: vi.fn().mockReturnValue('main'),
+  };
+});
+
+vi.mock('simple-git', async () => {
+  const actual = await vi.importActual('simple-git');
+  return {
+    ...actual,
+    simpleGit: () => ({
+      log: ({ maxCount }: { maxCount: number } = { maxCount: 1 }) =>
+        Promise.resolve({
+          all: [
+            { hash: 'commit-6' },
+            { hash: 'commit-5' },
+            { hash: 'commit-4' },
+            { hash: 'commit-3' },
+            { hash: 'commit-2' },
+            { hash: 'commit-1' },
+          ].slice(-maxCount),
+        }),
+    }),
+  };
+});
+
+describe('history-command', () => {
+  it('should have correct default arguments if no cli options are provided', async () => {
+    const result = await yargsCli(
+      ['history', '--config=/test/code-pushup.config.ts'],
+      {
+        ...DEFAULT_CLI_CONFIGURATION,
+        commands: [yargsHistoryCommandObject()],
+      },
+    ).parseAsync();
+
+    expect(result).toEqual(
```

Comment:
Please create separate yargs integration tests. Unit tests of a given command only test its internal logic, not parsing of its options. That is done in `yargs-cli.integration.test.ts`. Same for the test below.

Reply:
No other command is tested like this, so I will stick to the current style and we can do a refactoring in another PR.

Reply:
🧪 Testing consistency: All commands are tested in the way I described. So please adjust the tests to be consistent with the adopted style, this one mixes two different things.
- Autorun has unit tests only with function calls [here](https://github.com/code-pushup/cli/blob/main/packages/cli/src/lib/autorun/autorun-command.unit.test.ts). Its options are tested **separately** in yargs [integration tests](https://github.com/code-pushup/cli/blob/main/packages/cli/src/lib/yargs-cli.integration.test.ts).
- Collect same [here](https://github.com/code-pushup/cli/blob/main/packages/cli/src/lib/collect/collect-command.unit.test.ts).
etc.